realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/AuthTests.java

```diff
@@ -145,16 +144,40 @@ public void onSuccess(SyncUser user) {
             @Override
             public void onError(ObjectServerError error) {
                 assertEquals(ErrorCode.INVALID_CREDENTIALS, error.getErrorCode());
-                throw new IllegalArgumentException("BOOM");
+                looperThread.testComplete();
```

Comment:
Point taken. I rewritten the test so it now test what the method actually says it does.